dsn-2018/bare_conf.tex

```diff
@@ -0,0 +1,1264 @@
+
+\textbf{TODO - audit introductory paragraphs}
+
+We seek to build new intuition about collisions in hash functions
+and particularly collisions in MD4. By developing techniques using logical
+cryptanalysis, we seek to complement and extend the results from
+differential cryptanalysis. In particular, we wish to use logical cryptanalysis
+to discuss the impacts of collisions in real-world systems. In this paper,
+we introduce and give justification for general techniques, and present some
+results of these techniques.
+
+To begin, real world systems are most vulnerable to two attacks: the preimage
+and second preimage. For systems relying on e.g., password validation via
+comparing two hashes, preimage attacks would allow inverting the hash,
+increasing the risk associated with losing a database of hashed passwords.
+On the other hand, systems relying on message validation, such as signature
+checks or file integrity checks would be easily broken with fast second
+preimage techniques. Both of these attacks are intractable, thus protecting
+real-world systems. However, much research has been devoted to collision
+attacks, producing efficient results.
+
+Thus, we seek to evaluate the relative difference between a collision
+attack and second preimage attacks, to give some measure of distance between
+them. Several high profile attacks against hash function usage have been
+made. For MD5, these include cloning a CA certificate \cite{Stevens2009}
+and creating malicious executables \cite{MD5Executables}. For SHA-1, a pair of
+PDFs with the same hash \cite{cryptoeprint:2017:190}. However, many of these
+techniques exploit not the hash, but the flexibility of the file format,
+relying instead on embedding arbitrary, collidable data and later checking for
+the presence of one of the matching blocks. This is especially obvious in the
+MD5 executable and SHA-1 PDF attacks. The work by M. Stevens to perform the
+CA Collision, on the other hand, was a chosen prefix attack extending
+previous work from 2007 \cite{Stevens2007}.
+
+Seeing the need to evaluate how much flexibility exists in a collision,
+we propose the following metrics for evaluating the utility of a collision
+class:
+\begin{enumerate}
+    \item The number of unique differentials a collision class has.
+    \item The number of unit-step neighbors a collision class has.
+    \item The maximum count of zeros in a the binary representation of a
+        colliding block (and likewise with ones).
+    \item Whether there exists a block which collides under multiple initial
+        values.
+    \item Whether or not zero, one, or both of the blocks in a collision may
+        be of ASCII values under any input block difference.
+\end{enumerate}
+Note that the first three are quantitative measures providing some measure of
+flexibility of a collision class, whereas the latter two are merely looking
+for a single witness for having the property. Depending on the scenario,
```

Comment:
I don't think you need the second "for" here. "for a single witness having the property"